src/courtroom.cpp

```diff
@@ -4476,75 +4479,60 @@ void Courtroom::refresh_iclog(bool skiplast)
 {
-  QString label_text_tr = QCoreApplication::translate(p_checkbox->metaObject()->className(), "%1").arg(p_checkbox->text());
-  // Get the pixel width of the string if it were p_checkbox's label
-  int label_px_width = p_checkbox->fontMetrics().boundingRect(label_text_tr).width();
+  int label_theme_width;
+  QString label_text_tr;
+  // Get the pixel width of the string if it were p_widget's label
+  int label_px_width = p_widget->fontMetrics().boundingRect(label_text_tr).width();
   QString filename = "courtroom_design.ini";
   // Get the width of the element as defined by the current theme
   pos_size_type design_ini_result = ao_app->get_element_dimensions(p_identifier, filename);
-  int label_theme_width = design_ini_result.width - 18; // the width of a checkbox on win10 + 5px of padding, should probably try to fetch the actual size
-  
-  //qInfo() << "I: Width of label text: " << label_px_width << "px. Theme's width: " << label_theme_width << "px.";
-  
-  // we can't do much with a 0-width widget, and there's no need to truncate if the theme gives us enough space
-  if (label_theme_width <= 0 || label_px_width < label_theme_width) {
-    qInfo() << "I: Truncation aborted for checkbox label " << label_text_tr << ", either theme width <= 0 or label width < theme width.";
-    return;
-  }
-  p_checkbox->setToolTip(label_text_tr);
 
-  QString truncated_label = label_text_tr;
-  int truncated_px_width = label_px_width;
-  while (truncated_px_width > label_theme_width && truncated_label != "…")
-  {
-    truncated_label.chop(2);
-    truncated_label.append("…");
-    //qInfo() << "I: Attempted to truncate label to string: " << truncated_label;
-    truncated_px_width = p_checkbox->fontMetrics().boundingRect(truncated_label).width();
+  QLabel* p_label = qobject_cast<QLabel *>(p_widget); // QLabel-specific operations
+  if (p_label != nullptr) {
+    label_theme_width = design_ini_result.width;
+    label_text_tr = QCoreApplication::translate(p_widget->metaObject()->className(), "%1").arg(p_label->text());
+    p_label->setToolTip(label_text_tr);
   }
-  p_checkbox->setText(truncated_label);
-  if (truncated_label == "…") {
-    // Since we're using the Unicode ellipsis instead of three periods, this shouldn't happen
-    // but if it does, it tends to cause infinite loops, so we break to be sure
-    qWarning() << "W: Potential infinite loop prevented: Checkbox label " << label_text_tr << "truncated to '…', so truncation was aborted.";
+  QCheckBox* p_checkbox = qobject_cast<QCheckBox *>(p_widget); // QCheckBox-specific operations
+  if (p_checkbox != nullptr) {
+    label_theme_width = design_ini_result.width - 18; // the width of a checkbox on win10 + 5px of padding, should probably try to fetch the actual size
+    label_text_tr = QCoreApplication::translate(p_widget->metaObject()->className(), "%1").arg(p_checkbox->text());
+    p_checkbox->setToolTip(label_text_tr);

```

Comment:
Both lines apply to both branches; extract from if-block.

Reply:
Done, good idea. I've used conditional operators to reduce the amount of ugly if/else as well, which looks very very odd when clang-format gets ahold of it.